clients/cfrestclient/rest_cloud_foundry_client_extended.go

```diff
@@ -1,65 +1,154 @@
+}
+
+func (c CloudFoundryRestClient) GetAppProcessStatistics(appGuid string) ([]models.ApplicationProcessStatistics, error) {
+	token, err := c.cliConn.AccessToken()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve access token: %s", err)
+	}
+	apiEndpoint, _ := c.cliConn.ApiEndpoint()
+
+	getAppProcessStatsUrl := apiEndpoint + "/v3/apps/" + appGuid + "/processes/web/stats"
+	body, err := sendRequestAndReadBody(getAppProcessStatsUrl, token)
+	if err != nil {
+		return nil, err
+	}
+	processStats, err := parseBody[models.AppProcessStatisticsResponse](body)
+	if err != nil {
+		return nil, err
+	}
+	return processStats.Resources, nil
+}
+
+func (c CloudFoundryRestClient) GetApplicationRoutes(appGuid string) ([]models.ApplicationRoute, error) {
+	token, err := c.cliConn.AccessToken()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve access token: %s", err)
+	}
+	apiEndpoint, _ := c.cliConn.ApiEndpoint()
+
+	getAppRoutesUrl := apiEndpoint + "/v3/apps/" + appGuid + "/routes"
+	return getPaginatedResources[models.ApplicationRoute](getAppRoutesUrl, token)
+}
+
+func (c CloudFoundryRestClient) GetServiceInstances(mtaId, currentSpaceGuid string) ([]models.CloudFoundryServiceInstance, error) {
+	token, err := c.cliConn.AccessToken()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve access token: %s", err)
+	}
+	apiEndpoint, _ := c.cliConn.ApiEndpoint()
+	mtaIdHash := md5.Sum([]byte(mtaId))
+	mtaIdHashStr := hex.EncodeToString(mtaIdHash[:])
+
+	getServicesUrl := apiEndpoint + "/v3/service_instances?fields[service_plan]=guid,name,relationships.service_offering&" +
+		"fields[service_plan.service_offering]=guid,name&space_guids=" + currentSpaceGuid +
+		"&label_selector=mta_id=" + mtaIdHashStr
+	return getPaginatedResourcesWithIncluded(getServicesUrl, token, buildServiceInstance)
+}
+
+func (c CloudFoundryRestClient) GetServiceBindings(serviceName string) ([]models.ServiceBinding, error) {
+	token, err := c.cliConn.AccessToken()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve access token: %s", err)
+	}
+	apiEndpoint, _ := c.cliConn.ApiEndpoint()
+
+	getServiceBindingsUrl := apiEndpoint + "/v3/service_credential_bindings?type=app&include=app&service_instance_names=" + serviceName
+	return getPaginatedResourcesWithIncluded(getServiceBindingsUrl, token, buildServiceBinding)
 }
 
 func (c CloudFoundryRestClient) getSharedDomainsInternal(cloudFoundryUrlElements CloudFoundryUrlElements) (*models.CloudFoundryResponse, error) {
 	params := &operations.GetSharedDomainsParams{
 		Page:           cloudFoundryUrlElements.Page,
 		ResultsPerPage: cloudFoundryUrlElements.ResultsPerPage,
-		OrderDirection: cloudFoundryUrlElements.OrderDirection,
 		Context:        context.TODO(),
 	}
 	token, err := c.TokenFactory.NewToken()
 	if err != nil {
 		return nil, baseclient.NewClientError(err)
 	}
 
+	//TODO filter by domains that have an owning organization
```

Comment:
This still uses v2 right? Shouldn't be refactored?

Reply:
It is refactored. Look at the "hidden" swagger-generated files. It uses v3 now.

Reply:
Ok, but this todo should be completed